### PR TITLE
docs: improve github auth providers

### PR DESCRIPTION
**Pull request by vinzscam**
## Hey, I just made a Pull Request!

Improved the Github auth docs, specifying only a single resolver in the example.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/auth/github/provider.md [hunk 1]
```diff
@@ -43,13 +43,12 @@ auth:
...
-            # typically you would pick one of these
-            - resolver: emailMatchingUserEntityProfileEmail
-            - resolver: emailLocalPartMatchingUserEntityName
+            # Matches the GitHub username with the Backstage user entity name.
+            # See https://backstage.io/docs/auth/github/provider#resolvers for more resolvers.
             - resolver: usernameMatchingUserEntityName
 ```
 
-The GitHub provider is a structure with three configuration keys:
+The GitHub provider is a structure with five configuration keys:
```
Comment (awanlin) [inline docs/auth/github/provider.md:51]:
```suggestion
The GitHub provider is a structure with these configuration keys:
```
That way we avoid having to update this number

**Review by awanlin — APPROVED** [review]
Thanks @vinzscam, one small nit but doesn't seem worth blocking this

**Review by Rugvip — APPROVED** [review]
Nice! 👍, :shipit:

**Review by camilaibs — APPROVED** [review]
🎉

**Pull request merged**